core/lib/zksync_core/src/metadata_calculator/mod.rs

```diff
@@ -123,6 +125,27 @@ impl MetadataCalculator {
         self.health_updater.subscribe()
     }
 
+    /// Runs the Merkle tree API server on the specified address.
+    pub fn run_api_server(
```

Comment:
Thought: it is a bit odd to see `MetadataCalculator` to be responsible for creating an HTTP server. Probably, I would've expected some way to get a DB snapshot (or a Merkle Tree instantiated over a snapshot) from the MetadataCalculator, and see _an externally created_ server utilize this API of MetadataCalculator. I.e. metadata calculator just owns the tree, but not much more.

That being said, I don't necessarily propose to rewrite the code now, just thought that it's worth mentioning.